.github/workflows/e2e-tests.yml

```diff
@@ -98,12 +98,16 @@ jobs:
     - name: Run all tests
       if: matrix.gutenberg == 'no'
       working-directory: projects/plugins/jetpack/tests/e2e
-      run: yarn test-e2e
+      run: yarn test-e2e --testPathIgnorePatterns=search
 
     - name: Run only blocks tests with latest Gutenberg
       if: matrix.gutenberg == 'latest'
       working-directory: projects/plugins/jetpack/tests/e2e
       run: yarn test-e2e --testNamePattern=blocks
+    - name: Run only search tests
+      if: matrix.gutenberg == 'no' && ( contains(github.event.pull_request.labels.*.name, 'Search') || contains(github.event.pull_request.labels.*.name, 'Instant Search') )
```

Comment:
Do the tests run on all '[Status] ...' labels? Does it make sense to restrict to certain ones like `[Status] Needs Team Review`, `[Status] Needs Review` and `[Status] Ready to Merge`?

Reply:
It runs on any '[Status]...' labels..  The problem is that adding labels doesn't seem to trigger the workflow even it seems it should according the workflow setting and the document:  https://docs.github.com/en/actions/reference/events-that-trigger-workflows#pull_request

Still trying to figure this out. Any suggestion is appreciated.

Reply:
Can we confirm that the `github.event.pull_request` payload includes a `labels` field? It doesn't appear to be the case according to the docs linked above.

Reply:
Hi @jsnmoon Pull request webhook data structure is documented [here](https://docs.github.com/en/developers/webhooks-and-events/webhook-events-and-payloads#pull_request), where the labels should be accessed using pull_request.pull_request.labels.* .name. 
But the structure of pull_request event in workflow is not the same, where the syntax would be just pull_request.labels. * .name, which is not documented anywhere I can find.

Reply:
Resolving this since it appears to work as originally intended.

Reply:
As the current setup, E2E tests are triggered by synchronize event, and before labels are added, the E2E tests should all be sweet already, otherwise we have to fix the test before adding the Review labels. So it’s not necessary to test these labels.
@brbrr mentioned they would possibly do a ' a single workflow file by introducing a file-based test runner' meaning, the determine which test suite to run by the files changed, which will be a lot faster than running the whole suite.

Reply:
@kangzj: Are we intending to ship a file-based test runner with this PR?

Reply:
@jsnmoon no not in this PR. brbrr just said they possibly will do one - as it's global thing, he didn't expect me to do it.

Reply:
Gotcha! 👍

Reply:
The idea is to trigger search suites if search related files changed, to trigger block tests likewise and etc automatically